versions/3.1.0.md

```diff
@@ -2677,7 +2676,14 @@ components:
 
 When request bodies or response payloads may be one of a number of different schemas, a `discriminator` object can be used to aid in serialization, deserialization, and validation.  The discriminator is a specific object in a schema which is used to inform the consumer of the document of an alternative schema based on the value associated with it.
 
-When using the discriminator, _inline_ schemas will not be considered.
+The discriminator object is legal only when using one of the composite keywords `oneOf`, `anyOf`, `allOf`. Both _inline_ and references are valid child schemas.
+When a child schema is a reference, the discriminator property _MUST_ be present in the payload.
+A discriminator MAY act as a "hint" to shortcut validation and selection of the matching schema which may be a costly operation, depending on the complexity of the schema.
```

Comment:
I moved this sentence up in the section because it was in the middle of an example. Isn't this sentence always true regardless of the specific example that was provided? If so, I think it would make sense to put it outside the example section.

What does the word MAY mean for validation exactly? Who decides whether the discriminator is used as a hint or not? It would be good to be completely unambiguous such that there is no discrepancy between OAS validators.

What are the OAS validators supposed to do?

1. For _inline_ child schemas, do not use the discriminator as a hint and validate the payload using JSON schema validation. 
1. For _referenced_ child schemas, the discriminator MAY be used as a hint to quickly determine if the payload matches the child schema or not. If the payload does match the discriminator value, the payload MUST be validated against the JSON schema.

Reply:
How about `one or more of the composite keywords`
You can have a schema that includes allOf and oneOf where the payload conforms to both definitions, the allOf validation passes and those properties are loaded in as additional properties in the chosen oneOf schema.

Reply:
> How about `using one of the composite keywords` -> `using one or more of the composite keywords`
> You can have a schema that includes allOf and oneOf where the payload conforms to both definitions, the allOf validation passes and those properties are loaded in as additional properties in the chosen oneOf schema.

Thank you I have updated the sentence accordingly.